src/CodeGeneration/ApiGenerator/Domain/Code/CsharpNames.cs

```diff
@@ -1,26 +1,56 @@
 using System;
+using System.Net.Http;
+using System.Text.RegularExpressions;
 using CsQuery.ExtensionMethods.Internal;
+using Microsoft.Extensions.FileSystemGlobbing.Internal.PathSegments;
 
 namespace ApiGenerator.Domain 
 {
 	public class CsharpNames
 	{
 		public CsharpNames(string name, string endpointMethodName, string endpointNamespace)
 		{
-			Namespace = endpointNamespace.ToPascalCase();
+			Namespace = CreateCSharpNamespace(endpointNamespace);
 			if (CodeConfiguration.ApiNameMapping.TryGetValue(name, out var mapsApiMethodName))
-				MethodName = mapsApiMethodName;
-			else MethodName = $"{Namespace}{endpointMethodName.ToPascalCase()}";
+				ApiName = mapsApiMethodName;
+			else ApiName = endpointMethodName.ToPascalCase();
+
+			//if the api name starts with the namespace do not repeat it in the method name
+			MethodName = Regex.Replace(ApiName, $"^{Namespace}(\\w+)$", "$1");
```

Comment:
++ also need to use a dictionary for pluralisation (Indices / Index) or abbreviations (ILM / Index Lifecycle Management)

Reply:
(just seen abbreviations below!)